docker/Makefile

```diff
@@ -1,16 +1,28 @@
-IMAGE_NAME ?= mosaicml/composer
-BUILD_TYPE ?= base
-
-ifeq (${BUILD_TYPE}, base)
-	# Base dependency build tagged with 'latest'
-	IMAGE_TAG = latest
-else
-	# Otherwise just use the BUILD_TYPE
-	IMAGE_TAG = ${BUILD_TYPE}
-endif
+BASE_IMAGE ?= 'nvidia/cuda:11.3.1-cudnn8-runtime-ubuntu20.04'
+PYTHON_VERSION ?='3.9'
+CUDA_VERSION_TAG ?= 'cu113'
+PYTORCH_VERSION ?= '1.10.0'
+TORCHVISION_VERSION ?= '0.11.1'
+PYTORCH_TAG ?= mosaicml/pytorch
+COMPOSER_TAG ?= mosaicml/composer
+COMPOSER_EXTRA_DEPS ?= base
```

Comment:
[nit] Wondering if we should move these versions into a conditional similar to the one removed above?  Ideally we don't support more than a few images but right now already seems like we will have:
1. Initial released image we released with: Ubuntu 18.04, Python 3.8, PyTorch 1.9)
2. New image: Ubuntu 20.04, Python 3.9, PyTorch 1.10
3. Legacy image: Ubuntu 18.04?, Python 3.7, PyTorch 1.9?

Reply:
I added documentation for these variables. I would prefer to keep the documentation for the build matrix out of the Makefile, but agree this should be in the readme, once we have these images published and released on dockerhub.